Cython/Compiler/ExprNodes.py

```diff
@@ -3902,8 +3902,34 @@ def analyse_target_declaration(self, env):
         pass
 
     def analyse_as_type(self, env):
+        modifier = self.base.as_cython_attribute()
+        if modifier is not None and modifier in ('const', 'volatile'):
+            # const[base_type] or volatile[base_type]
+            is_const = modifier == 'const'
+            is_volatile = not is_const
+            base_type = self.index.analyse_as_type(env)
+            if base_type is None:
+                error(self.base.pos, f"invalid use of '{modifier}', argument is not a type")
+                return None
+            elif base_type.is_cv_qualified:
```

Comment:
This probably needs a test for `const[pointer[const[int]]]` which I believe should be valid. 

I suspect it might pass already but I think it's worth having a test for.

Reply:
It's currently `PointerType[cython.int]` because `pointer()` is a function and doesn't allow item access. Should we change `pointer` into a callable type class? That would avoid having two names for essentially the same thing.

I tried `const[PointerType[const[int]]]` and that's not currently accepted by this PR because Cython's type analysis doesn't know about `PointerType` yet. So now would be the time to fix the name.